image/factory.go

```diff
@@ -0,0 +1,27 @@
+package image
+
+import (
+	"github.com/buildpack/imgutil"
+	"github.com/docker/docker/client"
+	"github.com/google/go-containerregistry/pkg/authn"
+)
+
+type DefaultImageFactory struct {
```

Comment:
Wondering if the word "Default" is appropriate for this type... What about it makes it "default-ey"? 

Side note: I see the convenience of the `NewImage` method, but aside from that, is this really buying us much?